src/components/organisms/AssetActions/Compute/index.tsx

```diff
@@ -169,21 +169,33 @@ export default function Compute({
     ) {
       algorithmSelectionList = []
     } else {
-      const gueryResults = await queryMetadata(
+      const queryResults = await queryMetadata(
         getQuerryString(
           computeService.attributes.main.privacy.publisherTrustedAlgorithms,
           ddo.chainId
         ),
         source.token
       )
-      setDdoAlgorithmList(gueryResults.results)
-      const datasetComputeService = ddo.findServiceByType('compute')
-      algorithmSelectionList = await transformDDOToAssetSelection(
-        undefined,
-        gueryResults.results,
-        [],
-        newCancelToken()
+      // verify if the algorithms in the trusted list share the same endpoint of the dataset
+      const sameProviderEndpointAlgorithms = queryResults.results.filter(
+        (algo) =>
+          (
+            algo.findServiceByType('compute') ||
+            algo.findServiceByType('access')
+          )?.serviceEndpoint === computeService.serviceEndpoint
       )
+      setDdoAlgorithmList(sameProviderEndpointAlgorithms)
+
+      if (sameProviderEndpointAlgorithms.length === 0) {
+        algorithmSelectionList = []
+      } else {
+        algorithmSelectionList = await transformDDOToAssetSelection(
+          undefined,
+          sameProviderEndpointAlgorithms,
+          [],
+          newCancelToken()
+        )
+      }
     }
```

Comment:
suggestion(blocking): This looks like there is a cleaner way to set this value. Think of something like this (there might be even better ways using `concat`): 

```javascript
if (sameProviderEndpointAlgorithms.length > 0) {
      algorithmSelectionList = await transformDDOToAssetSelection(
          undefined,
          sameProviderEndpointAlgorithms,
          [],
          newCancelToken()
        )
}
```

Reply:
I ended up refactoring the whole function, it looks better to me now